File: src/test/java/no/nav/syfo/util/OidcTestHelper.java
package no.nav.syfo.util;

import com.nimbusds.jwt.JWTClaimsSet;
import com.nimbusds.jwt.SignedJWT;
import no.nav.security.token.support.core.context.TokenValidationContext;
import no.nav.security.token.support.core.context.TokenValidationContextHolder;
import no.nav.security.token.support.core.jwt.JwtToken;
import no.nav.security.token.support.test.JwtTokenGenerator;

import java.text.ParseException;
import java.util.HashMap;
import java.util.Map;

import static no.nav.syfo.security.OIDCIssuer.AZURE;
import static no.nav.syfo.security.OIDCIssuer.VEILEDERAZURE;

public class OidcTestHelper {

    public static void loggInnVeilederMedAzure(TokenValidationContextHolder oidcRequestContextHolder, String veilederIdent) throws ParseException {
        JWTClaimsSet claimsSet = JWTClaimsSet.parse("{\"NAVident\":\"" + veilederIdent + "\"}");
        SignedJWT jwt = JwtTokenGenerator.createSignedJWT(claimsSet);
        settOIDCValidationContext(oidcRequestContextHolder, jwt, AZURE);
    }

    public static void logInVeilederWithAzure2(TokenValidationContextHolder oidcRequestContextHolder, String veilederIdent) throws ParseException {
        JWTClaimsSet claimsSet = JWTClaimsSet.parse("{ }");
        SignedJWT jwt = JwtTokenGenerator.createSignedJWT(claimsSet);
        settOIDCValidationContext(oidcRequestContextHolder, jwt, VEILEDERAZURE);
    }

    public static void loggUtAlle(TokenValidationContextHolder oidcRequestContextHolder) {
        oidcRequestContextHolder.setTokenValidationContext(null);
    }

    private static void settOIDCValidationContext(TokenValidationContextHolder tokenValidationContextHolder, SignedJWT jwt, String issuer) {
        JwtToken jwtToken = new JwtToken(jwt.serialize());
        Map<String, JwtToken> issuerTokenMap = new HashMap<>();
        issuerTokenMap.put(issuer, jwtToken);
        TokenValidationContext tokenValidationContext = new TokenValidationContext(issuerTokenMap);
        tokenValidationContextHolder.setTokenValidationContext(tokenValidationContext);

    }

}

File: src/test/kotlin/no/nav/syfo/testhelper/OidcTestHelper.kt
package no.nav.syfo.testhelper

import com.nimbusds.jwt.JWTClaimsSet
import com.nimbusds.jwt.SignedJWT
import no.nav.security.token.support.core.context.TokenValidationContext
import no.nav.security.token.support.core.context.TokenValidationContextHolder
import no.nav.security.token.support.core.jwt.JwtToken
import no.nav.security.token.support.test.JwtTokenGenerator
import no.nav.syfo.api.auth.OIDCIssuer.VEILEDERAZURE
import java.text.ParseException
import java.util.*

object OidcTestHelper {
    @JvmStatic
    @Throws(ParseException::class)
    fun logInVeilederWithAzure2(
        oidcRequestContextHolder: TokenValidationContextHolder,
        consumerClientId: String = "",
        veilederIdent: String
    ) {
        val claimsSet = JWTClaimsSet.parse("{ \"azp\": \"$consumerClientId\", \"NAVident\": \"$veilederIdent\"}")
        val jwt = JwtTokenGenerator.createSignedJWT(claimsSet)
        settOIDCValidationContext(oidcRequestContextHolder, jwt, VEILEDERAZURE)
    }

    @JvmStatic
    fun loggUtAlle(oidcRequestContextHolder: TokenValidationContextHolder) {
        oidcRequestContextHolder.tokenValidationContext = null
    }

    private fun settOIDCValidationContext(tokenValidationContextHolder: TokenValidationContextHolder, jwt: SignedJWT, issuer: String) {
        val jwtToken = JwtToken(jwt.serialize())
        val issuerTokenMap: MutableMap<String, JwtToken> = HashMap()
        issuerTokenMap[issuer] = jwtToken
        val tokenValidationContext = TokenValidationContext(issuerTokenMap)
        tokenValidationContextHolder.tokenValidationContext = tokenValidationContext
    }
}
